src/Umbraco.Cms.ManagementApi/Factories/ExamineIndexViewModelFactory.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Examine;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Infrastructure.Examine;
+using Umbraco.Cms.ManagementApi.ViewModels.ExamineManagement;
+using Umbraco.New.Cms.Infrastructure.Services;
+
+namespace Umbraco.Cms.ManagementApi.Factories;
+
+public class ExamineIndexViewModelFactory : IExamineIndexViewModelFactory
+{
+    private readonly IIndexDiagnosticsFactory _indexDiagnosticsFactory;
+    private readonly IIndexRebuilder _indexRebuilder;
+    private readonly ITemporaryIndexingService _temporaryIndexingService;
+
+    public ExamineIndexViewModelFactory(IIndexDiagnosticsFactory indexDiagnosticsFactory, IIndexRebuilder indexRebuilder, ITemporaryIndexingService temporaryIndexingService)
+    {
+        _indexDiagnosticsFactory = indexDiagnosticsFactory;
+        _indexRebuilder = indexRebuilder;
+        _temporaryIndexingService = temporaryIndexingService;
+    }
+
+    public ExamineIndexViewModel Create(IIndex index)
+    {
+        if (_temporaryIndexingService.Detect(index.Name))
+        {
+            throw new OperationCanceledException("The index is still rebuilding, so could not get it");
```

Comment:
Instead of throwing an exception here, maybe return an `ExamineIndexViewModel` that has `"Rebuilding"` as  `HealthStatus` and `0` as `DocumentCount` and `FieldCount` ?